app/apps/addresses/views.py

```diff
@@ -48,28 +48,27 @@ class AddressViewSet(ViewSet, GenericAPIView, PermitDetailsMixin):
     )
     def residents_by_bag_id(self, request, bag_id):
         address = self.queryset.filter(bag_id=bag_id).first()
-        if not address:
+        address_designation_id = None
+        if address:
             try:
-                bag_data = do_bag_search_id(bag_id)
-                result = bag_data.get("results", [])[0]
-                address = {
-                    "postal_code": result.get("postcode", ""),
-                    "number": result.get("huisnummer", ""),
-                    "suffix": result.get("bag_toevoeging", ""),
-                    "suffix_letter": result.get("bag_huisletter", ""),
-                }
+                bag_data = do_bag_search_number_designations_id(bag_id)
+                results = bag_data.get("_embedded", {}).get("nummeraanduidingen", [])
+                for result in results:
+                    if result.get("huisnummer", "") == address.number:
+                        break
+                else:
+                    result = None
+
```

Comment:
Suggestie voor het hele blok (56-60):
`next((result for result in results if x == address.number), {})`

Maar dan geen `results` gebruiken maar een duidelijkere naam